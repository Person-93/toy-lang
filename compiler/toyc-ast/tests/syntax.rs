use anyhow::Result;
use chumsky::Parser;
use test_util::{assert_debug_snapshot, snapshots};
use toyc_ast::{ast, tokens::TokenIter};

fn main() -> Result<()> {
  snapshots(|text| {
    let tokens = TokenIter::new(text).into_stream();
    assert_debug_snapshot!(ast::parse::file().parse_recovery(tokens));
  })
}
