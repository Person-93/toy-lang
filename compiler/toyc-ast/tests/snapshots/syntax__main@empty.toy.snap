---
source: compiler/toyc-ast/tests/syntax.rs
assertion_line: 9
expression: "ast::parse::file().parse_recovery(tokens)"

---
(
    Some(
        File {
            attrs: [],
            items: [],
            span: Span {
                lo: 0,
                hi: 0,
            },
        },
    ),
    [],
)
